Client/Experiments/Experiments.swift

```diff
@@ -0,0 +1,165 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import MozillaAppServices
+import Shared
+import XCGLogger
+
+private let log = Logger.browserLogger
+
+/// `Experiments` is the main entry point to use the `Nimbus` experimentation platform in Firefox for iOS.
+///
+/// This class is a application specific holder for a the singleton `NimbusApi` class.
+///
+/// It is needed to be initialized early in the startup of the app, so a lot of the heavy lifting of calculating where the
+/// database should live, and deriving the `Remote Settings` URL for itself. This should be done with the
+/// `initialize(with:,firstRun:)` method.
+///
+/// Most usage with be made of `Nimbus` by feature developers who wish to make decisions about how
+/// to configure their features.
+///
+/// This should be done with the `withExperiment(featureId:)` method.
+/// ```
+/// button.text = Exeriments.shared.withExperiment(featureId: .submitButton) { branchId in
+///    switch branchId {
+///      ExperimentBranch.treatment -> return "Ok then"
+///      else -> return "OK"
+///    }
+/// }
+/// ```
+///
+/// Possible values for `featureId` correspond to the application features under experiment, and are
+/// enumerated  in the `FeatureId` `enum` in `ExperimentConstants.swift`.
+///
+/// Branches are left as `String`s as they are an unbounded set of values, but commonly used
+/// constants are also defined in `ExperimentConstants`.
+///
+/// The server components of Nimbus are: `RemoteSettings` which serves the experiment definitions to
+/// clients, and `Experimenter`, which is the user interface for creating and administering experiments.
+///
+/// Rust errors are not expected, but will be reported via Sentry.
+enum Experiments {
+    static var dbPath: String? {
+        let profilePath: String?
+        if AppConstants.IsRunningTest || AppConstants.IsRunningPerfTest {
+            profilePath = (UIApplication.shared.delegate as? TestAppDelegate)?.dirForTestProfile
+        } else {
+            profilePath = FileManager.default.containerURL(
+                forSecurityApplicationGroupIdentifier: AppInfo.sharedContainerIdentifier
+            )?
+                .appendingPathComponent("profile.profile")
+                .path
+        }
+        let dbPath = profilePath.flatMap {
+            URL(fileURLWithPath: $0).appendingPathComponent("nimbus.db").path
+        }
+
+        if let dbPath = dbPath, Logger.logPII {
+            log.info("Nimbus database: \(dbPath)")
+        }
+        return dbPath
+    }
```

Comment:
This feels like it should be utility function someplace (it's all `static`), but I couldn't find one.

Reply:
I've been worried about extensions. 

I had considered working with extensions to be part of another, as yet defined epic, once there was a demand for it. It will require some fundamental work in Nimbus SDK, e.g. understanding accessing `rkv` across process boundaries, running rkv in different processes to access the same file, or running completely different instances of nimbus and database, that only share ids (e.g. in the keychain).

That Glean doesn't support extensions yet, is just yet another reason for Nimbus not worry about them right now.